### ISSUE TITLE
Implement KHR_materials_anisotropy glTF extension support

### ISSUE THREAD

**Issue by mvaligursky** [playcanvas/engine#6135]
https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Khronos/KHR_materials_anisotropy/README.md

![image](https://github.com/playcanvas/engine/assets/59932779/1758811a-8a27-49b3-b6c2-037313732245)

test models: https://github.com/cx20/gltf-test/tree/master/tutorialModels/AnisotropyBarnLamp

**Comment by emadurandal** [comment]
Hello, I was invited by Mr. Eastcott to contribute PlayCanvas.
https://x.com/willeastcott/status/1920844676648538372

[My library](https://github.com/actnwit/RhodoniteTS) supports KHR_materials_anisotropy, so maybe I can help you.

How can we start collaborating?

**Comment by mvaligursky** [comment]
Hi .. nice to hear you're keen to tackle this one, that'd be fantastic!

The way this probably needs to be done:
- add a new example, perhaps based on [this](https://playcanvas.vercel.app/#/graphics/clear-coat) but loading the anisotropic test model
- figure out what properties we need to add to `StandardMaterial` and add them.
- update GlbParser to parse the extension and set those properties on the material, similarly to other extensions
- update StandardMaterialOptionsBuilder to convert those material settings to options for the shader generator
- update LitShader to convert those options to defines that the shader consumes
- and then update shader chunks / add new chunks as needed, using those defines, to add the shader functionality. In GLSL at least (we can then convert to WGSL), or both.

I skipped lots of details, but this is a rough overview.

**Comment by emadurandal** [comment]
Thank you for the guidance.
First, I will proceed to understand how those PlayCanvas shaders work.

**Comment by emadurandal** [comment]
I'm currently working on understanding the code with reference to Clearcoat's implementation.
However, since the related code has been written for various files (including around lights, etc.), I am wondering where to start.
I need time to explore this a little.

**Comment by emadurandal** [comment]
@mvaligursky 

I learned that StandardMaterial already has a property 'anisotropy'. 
https://github.com/playcanvas/engine/blob/0e1c148decd4a86565c15074690a1d12e80bfbc2/src/scene/materials/standard-material.js#L134-L141

Also, it seems that there is already a shader function in GGX that takes Anisotropy into account.
https://github.com/playcanvas/engine/blob/main/src/scene/shader-lib/chunks-glsl/lit/frag/lightSpecularAnisoGGX.js

These seem to have been defined a long time ago, should they be used in this time implementation?

**Comment by mvaligursky** [comment]
Yes we definitely have a version of anisotropy. I'm not much familiar with it, but it's lacking properties compared to the glTF spec, and likely the implementation is non-matching. So this would need a review + adjustment or replacement, depending how close to the spec it is / can be made.

**Comment by emadurandal** [comment]
I looked at the GGX code below and it basically matches the algorithm described in the KHR_materials_anisotropy specification, but it does not seem to take into account the rotation (direction of anisotropy) property.
https://github.com/playcanvas/engine/blob/main/src/scene/shader-lib/chunks-glsl/lit/frag/lightSpecularAnisoGGX.js

A modification to this function to take the rotation property into account seems to solve the problem.

**Comment by emadurandal** [comment]
For the contents of the following file, not only is there no element for the direction of anisotropy, but the way bentNormal is calculated is slightly different from KHR_materials_anisotropy.
https://github.com/playcanvas/engine/blob/main/src/scene/shader-lib/chunks-glsl/lit/frag/reflDirAniso.js

If the contents of this getReflDir are modified to support KHR_materials_anisotropy, the anisotropic behavior may be different from the previous implementation, even with rotation set to the initial value of 0 (no rotation).
Is this acceptable?

**Comment by emadurandal** [comment]
I created a draft PR https://github.com/playcanvas/engine/pull/7668 . I do not expect this to be accepted as is, but I hope it will serve as a starting point for discussion.

**Comment by mvaligursky** [comment]
fixed in https://github.com/playcanvas/engine/pull/7668
